src/scan/postgres_scan.cpp

```diff
@@ -422,6 +422,8 @@ PostgresScanGlobalState::PostgresScanGlobalState(Snapshot _snapshot, Relation _r
     : snapshot(_snapshot), rel(_rel), table_tuple_desc(RelationGetDescr(rel)), count_tuples_only(false),
       output_columns(), total_row_count(0), registered_local_states(0), scan_query(),
       table_reader_global_state(nullptr), duckdb_scan_memory_ctx(nullptr), max_threads(1) {
+	PostgresScopedStackReset scoped_stack_reset;
+	std::lock_guard<std::recursive_mutex> lock(GlobalProcessLock::GetLock());
```

Comment:
Why are these changes suddenly needed? (as well as the one below in `PostgresScanFunction`) It seems like these additions could very well be the cause for the test failures in CI.